automation/tests/provisioning/pg-ha_test_2/pg-ha_test.go

```diff
@@ -0,0 +1,212 @@
+	if err = suite.tsm.TearDown(suite.ctx, setupTypes, suite.clientset, suite.v1alpha1ClientSet, suite.T()); err != nil {
+		logger.Printf("%s! %s\n", errBaseMsg, err)
+		suite.T().FailNow()
+	}
+
+	logger.Println("HA TearDownSuite() completed!")
+}
+
+// This will run right before the test starts and receives the suite and test names as input
+func (suite *PostgresProvisioningHighAvailabilityTestSuite) BeforeTest(suiteName, testName string) {
+	util.GetLogger(suite.ctx).Printf("******************** RUNNING HA TEST %s %s ********************\n", suiteName, testName)
+}
+
+// This will run after test finishes and receives the suite and test names as input
+func (suite *PostgresProvisioningHighAvailabilityTestSuite) AfterTest(suiteName, testName string) {
+	util.GetLogger(suite.ctx).Printf("******************** END HA TEST %s %s ********************\n", suiteName, testName)
+}
+
+// Tests if provisioning is succesful by checking if database status is 'READY'
+func (suite *PostgresProvisioningHighAvailabilityTestSuite) TestProvisioningSuccess() {
+	logger := util.GetLogger(suite.ctx)
+
+	databaseResponse, err := suite.tsm.GetDatabaseOrCloneResponse(suite.ctx, suite.setupTypes, suite.clientset, suite.v1alpha1ClientSet)
+	if err != nil {
+		logger.Printf("Error: TestProvisioningSuccess() failed! %v", err)
+	} else {
+		logger.Println("Database response retrieved.")
+	}
+
+	assert := assert.New(suite.T())
+	assert.Equal(common.DATABASE_CR_STATUS_READY, databaseResponse.Status, "The database status should be ready.")
+}
+
+// Tests if app is able to connect to database via GET request
+func (suite *PostgresProvisioningHighAvailabilityTestSuite) TestAppConnectivity() {
+	logger := util.GetLogger(suite.ctx)
+
+	resp, err := suite.tsm.GetAppResponse(suite.ctx, suite.setupTypes, suite.clientset, automation.POSTGRES_SI_PROVISONING_LOCAL_PORT)
+	if err != nil {
+		logger.Printf("Error: TestAppConnectivity failed! %v", err)
+	} else {
+		logger.Println("App response retrieved.")
+	}
+
+	assert := assert.New(suite.T())
+	assert.Equal(200, resp.StatusCode, "The response status should be 200.")
+}
+
+// Tests if creation of time machine is succesful
+func (suite *PostgresProvisioningHighAvailabilityTestSuite) TestTimeMachineSuccess() {
+	logger := util.GetLogger(suite.ctx)
+	assert := assert.New(suite.T())
+
+	if suite.setupTypes.Database.Spec.Instance.TMInfo.SLAName == "" || suite.setupTypes.Database.Spec.Instance.TMInfo.SLAName == "NONE" {
+		logger.Println("No time machine specified, test automatically passing.")
+		return
+	}
+
+	tm, err := suite.tsm.GetTimemachineResponseByDatabaseId(suite.ctx, suite.setupTypes, suite.clientset, suite.v1alpha1ClientSet)
+	if err != nil {
+		logger.Printf("Error: TestTimeMachineSuccess() failed! %v", err)
+		assert.FailNow("Error: TestTimeMachineSuccess() failed! %v", err)
+	} else {
+		logger.Println("Timemachine response retrieved.")
+	}
+
+	err = util.CheckTmInfo(suite.ctx, suite.setupTypes.Database, &tm)
```

Comment:
I wonder why its a pointer, maybe if we don't dereference twice?

```suggestion
	err = util.CheckTmInfo(suite.ctx, suite.setupTypes.Database, tm)
```